pkg/cluster/cluster.go

```diff
@@ -100,6 +104,11 @@ type Options struct {
 	// will only hold objects from the desired namespace.
 	Namespace string
 
+	// HTTPClient is the http client that will be used to create the default
+	// Cache and Client. If not set the rest.HTTPClientFor function will be used
+	// to create the http client.
+	HTTPClient *http.Client
```

Comment:
Does this need to be exporter here? Why?

Reply:
@vincepri I have to be able to access the client here: https://github.com/kubernetes-sigs/controller-runtime/pull/2122/files#diff-d500fbd6a2aa620607ca5e2a7c3ac4f1a4c82309d1a549561e92abfcb18f2f0eR368-R382